lib/apple/index.js

```diff
@@ -50,18 +51,26 @@ function parseResult(result) {
 	var transactionId = getReceiptFieldValue(result.receipt, 'transaction_id');
 
 	/* jshint camelcase:false */
-	if (result.hasOwnProperty('latest_receipt_info') && result.latest_receipt_info[0]) {
-		latestReceiptInfo = result.latest_receipt_info.sort(function (a, b) {
-			return parseInt(a.transaction_id, 10)  - parseInt(b.transaction_id, 10);
-		});
-
-		productId = latestReceiptInfo[latestReceiptInfo.length - 1].product_id;
-		transactionId = latestReceiptInfo[latestReceiptInfo.length - 1].transaction_id;
+	if (result.hasOwnProperty('latest_receipt_info')) {
+		if(Array.isArray(result.latest_receipt_info) && result.latest_receipt_info.length > 1) {
```

Comment:
If length is 1, it seems the `else` clause below will not treat it well, I think?

Reply:
Yeah unfortunately, it can be both an array of 1, or an object that isn't an array. So this code fails in that instance. I added the following to handle this case:

else if(Array.isArray(result.latest_receipt_info) && result.latest_receipt_info.length == 1) {
            latestReceiptInfo = result.latest_receipt_info;
            productId = latestReceiptInfo[latestReceiptInfo.length - 1].product_id;
            transactionId = latestReceiptInfo[latestReceiptInfo.length - 1].transaction_id;
        }

Reply:
@ronkorving & @clarky07 That's a good point about the single entry array. I never saw that in my testing since if it was an object for single entry or an array with more than 1. But I'll update my PR just in case that turns into the what is done in the future.